synapse/pipeline.yml

```diff
@@ -527,3 +527,34 @@ steps:
 #
 #  - label: Trigger webhook
 #    command: "curl -k https://coveralls.io/webhook?repo_token=$COVERALLS_REPO_TOKEN -d \"payload[build_num]=$BUILDKITE_BUILD_NUMBER&payload[status]=done\""
+
+  ################################################################################
+  #
+  # Complement Test Suite
+  #
+  ################################################################################
+
+  - command:
+      - "docker build -t matrixdotorg/synapse:latest -f docker/Dockerfile ."
+      - "wget -N https://github.com/matrix-org/complement/archive/master.tar.gz && tar -xzf master.tar.gz"
+      - "docker build -t complement-synapse -f complement-master/dockerfiles/Synapse.Dockerfile complement-master/dockerfiles"
+      - "cd complement-master && COMPLEMENT_BASE_IMAGE=complement-synapse:latest go test -v ./tests"
```

Comment:
can we have a comment describing what this is doing and why?

For instance: why do we pull a docker image of `complement:latest`, but then download a separate tarball, unpack it, and run that?

why do we build two "synapse" docker images?

Reply:
I've added some comments, the process is indeed a bit complicated. I think it could be cleaned up actually, perhaps by mounting the Synapse checkout instead of building one image from another. But not sure whether we want to do that atm.

I'm also not sure how long this will take to run on our build infra. If it's longer than Sytest is currently then I'd probably advocate for putting in the work to mount instead of building a Synapse container from scratch.